tests/Fixer/Alias/SetTypeToCastFixerTest.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Fixer\Alias;
+
+use PhpCsFixer\Tests\Test\AbstractFixerTestCase;
+
+/**
+ * @author SpacePossum
+ *
+ * @internal
+ *
+ * @covers \PhpCsFixer\Fixer\Alias\SetTypeToCastFixer
+ */
+final class SetTypeToCastFixerTest extends AbstractFixerTestCase
+{
+    /**
+     * @param string      $expected
+     * @param null|string $input
+     *
+     * @dataProvider provideFixCases
+     */
+    public function testFix($expected, $input = null)
+    {
+        $this->doTest($expected, $input);
+    }
+
+    public function provideFixCases()
+    {
+        return [
+            [
+                '<?php (array)    $foo;',
+                '<?php settype  (  $foo  , \'array\');',
+            ],
+            [
+                '<?php (bool)    $foo;',
+                '<?php settype  (  $foo  , "Bool");',
+            ],
+            [
+                '<?php (bool)    $foo;',
+                '<?php settype  (  $foo  , "boolean");',
+            ],
+            [
+                '<?php (float) $foo;',
+                '<?php settype($foo, "double");',
+            ],
+            [
+                '<?php (float) $foo;',
+                '<?php settype($foo, "float");',
+            ],
+            'int full caps' => [
+                '<?php (int) $foo;',
+                '<?php settype($foo, "INT");',
```

Comment:
should it be lower casing them? or is that a job for `lowercase_cast`?

Reply:
this is the input, new cast is always in lowercase